src/main/java/hotelserviceapp/sources/Rooms.java

```
@@ -54,24 +52,25 @@ public void maintainRoom(LocalDate newMaintenanceDate) {
 	 * that is to be booked.
 	 * It later traverses the booking set and checks whether the selected dates are already booked by checking whether the
 	 * starting/end dates are already present in the booking set.
-	 * Returns an asnwer on whether the booking was successful.
+	 * Returns room number on successful booking , and negative value on failure.
 	 *
 	 * @param guestEGN       newGuestEGN is a string type variable , containing the guest's EGN number.
 	 * @param fromDate       fromDate is a LocalDate type variable which contains the requested booking's starting date.
 	 * @param toDate         toDate is a LocalDate type variable which contains the requested booking's end date .
 	 * @param roomToBeBooked roomToBeBooked is an object which represents the room that is being booked.
 	 * @param numberOfDays   numberOfDays is an integer type variable which represents the number of days for which the room will be booked.
 	 */
-	public boolean createBooking(String guestEGN, LocalDate fromDate, LocalDate toDate, Rooms roomToBeBooked, int numberOfDays) {
+	public int createBooking(String guestEGN, LocalDate fromDate, LocalDate toDate, Rooms roomToBeBooked, int numberOfDays) {
 		Booking objectToBeBooked = new Booking();
 		Iterator<Booking> temporaryIterator = bookings.iterator();
 		objectToBeBooked.updateRoom(guestEGN, fromDate, toDate, roomToBeBooked, numberOfDays);
-
-		if (checkAvailability(objectToBeBooked, temporaryIterator)) {
-			bookings.add(objectToBeBooked);
-			return true;
+		if (!objectToBeBooked.isPresentIn(bookings)) {
+			if (checkAvailability(objectToBeBooked, temporaryIterator)) {
+				bookings.add(objectToBeBooked);
+				return roomNumber;
+			}
 		}
-		return false;
+		return -1;
```

let's throw an exception if it's not found